.gitbook/assets/ReadonlyREST-api.yml

```diff
@@ -0,0 +1,209 @@
+                  $ref: "#/components/examples/UpdateMainSettingsSuccessResponse"
+                UpdateMainSettingsFailedResponse:
+                  $ref: "#/components/examples/UpdateMainSettingsFailedResponse"
+        403:
+          description: The request was forbidden by ReadonlyREST's ACL
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ForbiddenRorResponse'
+              examples:
+                NoAclBlockWasMatchedForbiddenResponseExample:
+                  $ref: "#/components/examples/NoAclBlockWasMatchedForbiddenResponseExample"
+                ForbidBlockMatchedResponseExample:
+                  $ref: "#/components/examples/ForbidBlockMatchedResponseExample"
+                TestSettingsNotConfiguredForbiddenResponseExample:
+                  $ref: "#/components/examples/TestSettingsNotConfiguredForbiddenResponseExample"
+                ImpersonationNotSupportedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotSupportedResponseExample"
+                ImpersonationNotAllowedResponseExample:
+                  $ref: "#/components/examples/ImpersonationNotAllowedResponseExample"
+                RorIsNotStartedYetResponseExample:
+                  $ref: "#/components/examples/RorIsNotStartedYetResponseExample"
+                RorFailedToStartResponseExample:
+                  $ref: "#/components/examples/RorFailedToStartResponseExample"
+                CannotSaveClusterWideSettingsOnFreeLicense:
+                  $ref: "#/components/examples/CannotSaveClusterWideSettingsOnFreeLicense"
+
+components:
+  securitySchemes:
+    http:
+      type: http
+      scheme: basic
+
+  schemas:
+    ForbiddenRorResponse:
+      type: object
+      required:
+        - error
+        - status
+      properties:
+        error:
+          type: object
+          required:
+            - type
+            - reason
+            - due_to
+          properties:
+            type:
+              type: string
+              enum:
+                - forbidden_response
+            reason:
+              type: string
+              enum:
+                - forbidden
+            due_to:
+              type: array
+              description: |
+                It's an array of causes why ReadonlyREST's ACL forbade the request. 
+                WARNING: Please notice, that one element array ES used to print as a string!!!
+              items:
+                type: string
+                enum:
+                  - OPERATION_NOT_ALLOWED
+                  - FORBIDDEN_BY_BLOCK
+                  - TEST_SETTINGS_NOT_CONFIGURED
+                  - IMPERSONATION_NOT_SUPPORTED
+                  - IMPERSONATION_NOT_ALLOWED
+                  - READONLYREST_NOT_READY_YET
+                  - READONLYREST_NOT_ENABLED
+                  - READONLYREST_FAILED_TO_START
+                  - cannotSaveClusterWideSettingsOnFreeLicense
```

Comment:
Why in a different style? 
Did you consider [this](https://readonlyrest.atlassian.net/browse/RORDEV-854?focusedCommentId=12807) instead?

Reply:
@sscarduzio please decide. IMO  both options are ok.

Reply:
Yeah change it, it looks foolish